conanfile.py

```diff
@@ -237,21 +237,7 @@ def imports(self):
             self.keep_imports = True
             self.copy("*.dll", "../Basilisk", "bin")
 
-    def generateMessageModules(self, originalWorkingDirectory):
```

Comment:
I think this file (http://hanspeterschaub.info/basilisk/Learn/makingModules/makingModules-2.html) should be updated to explain to the user when they need to do a clean build again to catch changes in the message edits or message deletions?

Reply:
The behaviour you described is currently the behaviour expected on other (non-`messaging`) packages, e.g. `ExternalModules`. That is, the build system will keep any previously built but renamed/deleted modules, and it is up to the developer to manually clean the build (e.g. with `--clean`). For example, building with `--pathToExternalModules external` will create `Basilisk.ExternalModules`, but if you then rebuild without this flag, that module will still exist unless the user runs a `--clean` build.

However, I have added an additional commit to maintain the desired auto-generation behaviour for the `messaging` package specifically. Now, auto-generated message files will get created in `dist3/messaging`, but only the *currently existing* messages will be copied from `dist3/messaging` -> `dist3/Basilisk/architecture/messaging`. This way,
- Previously built messages still exist in `dist3/messaging`, so they don't need to be rebuilt unless their dependencies change.
- Only messages which have an existing `*Payload.h` file will actually get copied into the final Basilisk package, so deleted messages won't be importable (although their compiled versions still exist in `dist3/messaging`).

This behaviour complicates things a bit, because the default behaviour of CMake is to leave previously built files alone until the user runs a `clean`, but I think the implementation is still okay. Please have a look and let me know if this is acceptable or not.

Reply:
you make a good point about the use of ``--clean``.  Maybe leave the original behavior, and we can just update the BSK documentation to explain when to use a `--clean` command?  

I don't mind the build behavior changing to be more consistent with how other elements are built (i.e. the modules folder, etc.). I just want to be explicit to the develop on what behavior to expect. :-)

Reply:
I've reverted the "Clean up stale message files..." change and went back to requiring `--clean`. I've added a caution to the documentation page you mentioned.

Note: I left the revert as an explicit revert commit in case we ever want to refer back to the implementation that only exports messages that actually exist. This might be useful in some long-term future build system changes (for reference, I'm considering possible ways to completely separate building of `ExternalModules` from building all of Basilisk).